### ISSUE TITLE
[Android] Pause on Bluetooth Disconnect from car

### ISSUE THREAD

**Issue by i-am-at0m** [advplyr/audiobookshelf-app#612]
### Is your feature request related to a problem? Please describe.
I often get in and out of my car, and in other apps playback pauses when Bluetooth disconnects. It does not do so on Audiobookshelf so I skipped an hour in my book while my phone was in my pocket playing on minimum volume because it kept playing after disconnecting. 

### Describe the solution you'd like
Pause on disconnect. Possibly play on connect if it's the active audio app but that's secondary IMO. 

### Describe alternatives you've considered
N/A

### Additional context
I am on a Pixel 7 Pro and using a Pixel Watch, so if this feature exists but only pauses when NO Bluetooth devices are connected, that could be a cause. But my watch isn't a playback device so that doesn't make much sense. Server version is v2.2.15, android app 0.9.62-beta.

**Comment by i-am-at0m** [comment]
Maybe this should be labeled a bug. I don't know what the intended behavior is.

**Comment by hobesman** [comment]
Mine pauses when Bluetooth disconnects, but I usually only have one Bluetooth device connected so I can't address that scenario with your watch.

**Comment by advplyr** [comment]
I just re-opened a similar issue #156 but I'm not sure if this is the same situation.

**Comment by i-am-at0m** [comment]
Looks like it, I'm just experiencing it with a car instead of headphones. I searched but was only looking for open issues and didn't see that one.

**Comment by advplyr** [comment]
Yeah I just re-opened it. That issue is also having to do with car bluetooth.
This may be specific to certain cars because it does pause in mine.

**Comment by i-am-at0m** [comment]
Yeah this is really killing usability for me. I caught it twice today but didn't catch it and lost another hour of my book, which isn't chaptered, so trying to find where I left off was rough.

Is there an option for me to turn on debug logs and do some controlled testing? Or use Android developer mode to do an info dump or something?

**Comment by advplyr** [comment]
Can you test this while only connected to one bluetooth device (your car)?

**Comment by advplyr** [comment]
Interesting the two reports of this issue are from Volkswagen bluetooth

**Comment by i-am-at0m** [comment]
> Can you test this while only connected to one bluetooth device (your car)?

Just tried it, put the watch in airplane mode, verified the phone only had one Bluetooth device connected, turned the car off and opened the door to power down the radio, and it didn't miss a beat and continued playing on my phone. 

> Interesting the two reports of this issue are from Volkswagen bluetooth

Yeah, wouldn't be the first time I've run into oddities with it + Android. It never reports the the correct track on Spotify, for example, it always tells me the previous track is the one currently playing. Doesn't do that with my wife's phone (iphone) but did with my Pixel 4a 5G and my new Pixel 7 Pro.

**Comment by advplyr** [comment]
Thanks for testing. Maybe there is an update you can get for the car or something?

**Comment by i-am-at0m** [comment]
> Thanks for testing. Maybe there is an update you can get for the car or something?

If I lived in Europe I could download them for free.

If I had a 2021 or newer, they would be OTA and also free.

If I had a 2015 or earlier, I could pay $125 for an update DVD via mail order (although when I did that on a 2012 the update wouldn't install. Waste of money.)

Since I have a 2016, the maps and software are on an SD card, and apparently they either don't trust people from the USA and Canada with that kind of technology or (more likely) they realized they can charge a premium for it and require updates to be installed at a dealer. And both VW dealers near me closed. 

TL;DR is no, I can't get an update.

**Comment by advplyr** [comment]
Is this still an issue for you?

**Comment by ufondu88** [comment]
I have the same issue. Sometimes, when I disconnect from my car's Bluetooth, the phone keeps playing the book on low volume and I might not notice for a few hours

**Comment by christianh104** [comment]
I have this issue as well so if there's anything I can do to help figure this out let me know, I'd love to help as this issue is a pain for me. This is happening with my car, a Polestar 2, and I do have a watch also connected to my phone. I noticed that when the car disconnects, Audiobookshelf does pause but a second or two later it starts playing again. I have not had this issue with any other media apps so I don't think it's an issue with the car or Android.

**Comment by pb-strawberries** [comment]
New to audiobookshelf and also having this same problem. When listening in my car '25 Chevy equinox, the app paused for a few moments when it disconnects from my car but then continues to play. This is different behavior than most of my other apps, audible and YT music stop playing and this is preferred.

**Comment by epipenRX** [comment]
Hi -- circling back on the secondary ask here ("possibly play on connect if it is the active audio app"). I tracked down the disconnect issue and submitted a fix in PR #1913 -- the root cause is that car head units send `KEYCODE_MEDIA_STOP` on power-off and the current code calls `closePlayback()` rather than `pause()`, destroying the session.

With the session kept alive by that fix, I am also building a follow-up enhancement: when the car disconnects, if a Bluetooth headset is already paired and connects within a few seconds, the app automatically resumes on it. Covers exactly the scenario you described.
